api.swagger.yaml

```diff
@@ -467,96 +576,803 @@ paths:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: |
+            Membership status returned
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/common-status-schema'
+        '400':
+          description: The UUID was malformed.
+        '404':
+          description: The group does not exist.
+      description: Check if the current user is a member of a group.
+  '/groups/{group_id}/members/{user_id}':
+    delete:
+      tags:
+        - groups
+      parameters:
+        - description: The Invotra UUID of the group.
+          in: path
+          name: group_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+        - description: The Invotra UUID of the user.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The user has been removed from the group.
+        '400':
+          description: The UUID was malformed.
+        '404':
+          description: |
+            The user or group does not exist, or the user was not a member
+            of the group.
+      description: Remove the user's membership of the group.
+    put:
+      tags:
+        - groups
+      parameters:
+        - description: The Invotra UUID of the group.
+          in: path
+          name: group_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+        - description: The Invotra UUID of the user.
+          in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+      responses:
+        '200':
+          description: The user's membership of this group was updated.
+        '201':
+          description: The user was added as a member of the group.
+        '400':
+          description: The UUID was malformed or the role type was invalid.
+        '404':
+          description: The user or group does not exist.
+      description: >-
+        Add a user to a group or update information about a user's existing
+        group membership.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/group-membership-common-schema'
+  /posts:
+    get:
+      tags:
+        - posts
+      parameters:
```

Comment:
again the external_id should either be required or we need a pager

Reply:
added